### ISSUE TITLE
Change default OpenTelemetry protocol to `http/protobuf`

### ISSUE THREAD

**Issue by brunobat** [quarkusio/quarkus#41522]
### Description

The default OTLP protocol must be changed from `grpc` to `http/protobuf` in order to align with the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.28.0/specification/protocol/exporter.md#specify-protocol).

There is something wrong with the span exporter when we change the protocol and the fix is too complex for https://github.com/quarkusio/quarkus/pull/41521

**Comment by geoand** [comment]
So I assume we need https://github.com/quarkusio/quarkus/pull/41521 in first?

**Comment by brunobat** [comment]
Yes we do and https://github.com/quarkusio/quarkus/pull/39032 as well, preferably.

**Comment by geoand** [comment]
👌

**Comment by jorsol** [comment]
Isn't `gRPC` generally more performant than `HTTP/Protobuf`?

Or is this change simply to adjust the default, which can still be explicitly configured to use `gRPC` if desired?

**Comment by brunobat** [comment]
@jamezp is going to take this one. Thanks!
